src/Razor/test/RazorLanguage.Test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -2838,6 +2839,78 @@ @using Microsoft.AspNetCore.Components.Web
             CompileToAssembly(generated);
         }
 
+        [Fact]
+        public void EventHandler_PreventDefault_StopPropagation_Minimized()
+        {
+            // Arrange
+
+            // Act
+            var generated = CompileToCSharp(@"
+@using Microsoft.AspNetCore.Components.Web
+<input @onclick:preventDefault @onclick:stopPropagation />");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+        }
+
+        [Fact]
+        public void EventHandler_PreventDefault_StopPropagation()
+        {
+            // Arrange
+
+            // Act
+            var generated = CompileToCSharp(@"
+@using Microsoft.AspNetCore.Components.Web
+<input @onfocus:preventDefault=""true"" @onclick:stopPropagation=""Foo"" @onfocus:stopPropagation=""false"" />
+@code {
+    bool Foo { get; set; }
+}");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+        }
+
+        [Fact]
+        public void EventHandler_WithDelegate_PreventDefault()
+        {
+            // Arrange
+
+            // Act
+            var generated = CompileToCSharp(@"
+@using Microsoft.AspNetCore.Components.Web
+<input @onfocus=""OnFocus"" @onfocus:preventDefault=""ShouldPreventDefault()"" />
+@code {
+    void OnFocus(FocusEventArgs e) { }
+
+    bool ShouldPreventDefault() { return false; }
+}");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+        }
+
+        [Fact]
+        public void EventHandler_PreventDefault_Duplicates()
+        {
+            // Arrange
+
+            // Act
+            var generated = CompileToCSharp(@"
+@using Microsoft.AspNetCore.Components.Web
+<input @onclick:preventDefault=""true"" @onclick:preventDefault />");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+        }
+
```

Comment:
This needs tests with generics both with and without type inference.

Reply:
I'm not sure what you mean here. Event handlers don't apply to custom components right? Things like preventDefault can apply only to DOM elements. Not sure how to use generics in them. Or am I misunderstanding the ask?